pkg/clusterclientstore/queue.go

```diff
@@ -38,8 +40,14 @@ func (s *Store) bindEventHandlers() {
 		},
 		Handler: kubecache.ResourceEventHandlerFuncs{
 			AddFunc: enqueueSecret,
-			UpdateFunc: func(_, newObj interface{}) {
-				enqueueSecret(newObj)
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				oldSecret := oldObj.(*corev1.Secret)
```

Comment:
I bet it would be great to type-cast like: oldSecret, ok := oldObj.(*corev1.Secret) to prevent a potential panic.

Reply:
I thought that was the job of the filter? If it's not, I'll fix that